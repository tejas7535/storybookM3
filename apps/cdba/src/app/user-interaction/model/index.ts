export * from './action.model';
export * from './interaction.model';
export * from './interaction-type.enum';
