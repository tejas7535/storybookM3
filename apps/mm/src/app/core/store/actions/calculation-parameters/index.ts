export * as CalculationParametersActions from './calculation-parameters.actions';
