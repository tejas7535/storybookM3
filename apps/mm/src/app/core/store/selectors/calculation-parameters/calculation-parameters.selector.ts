import { createSelector } from '@ngrx/store';

import { getCalculationParametersState } from '../../reducers';

export const getCalculationParameters = createSelector(
  getCalculationParametersState,
  (state) => state?.parameters || undefined
);
