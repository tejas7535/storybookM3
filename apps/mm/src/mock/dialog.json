{
  "pages": [
    {
      "id": "RSY_PAGE_BEARING_TYPE",
      "title": "RSY_BEARING_TYPE",
      "text": "dialog.bearing",
      "visible": {
        "value": true,
        "dataType": "boolean",
        "functionType": null,
        "expressionType": "CONSTANT",
        "ids": null,
        "id": null
      },
      "groups": [
        {
          "members": [
            {
              "id": "RSY_BEARING_TYPE",
              "text": "dialog.bearingType",
              "type": "list",
              "defaultValue": null,
              "abbreviation": "",
              "memberType": "Variable",
              "requiresUpdate": true,
              "requiresInitialization": false,
              "input": {
                "value": true,
                "dataType": "boolean",
                "functionType": null,
                "expressionType": "CONSTANT",
                "ids": null,
                "id": null
              },
              "listValues": [
                {
                  "id": 1,
                  "text": "dialog.bearingTypeListValues.radialSelfAligningRollerBearing",
                  "imageUrl": "https://mountingmanager-cae.schaeffler.com/api/Images/pero_unknown.bmp"
                },
                {
                  "id": 2,
                  "text": "dialog.bearingTypeListValues.selfAligningBallBearing",
                  "imageUrl": "https://mountingmanager-cae.schaeffler.com/api/Images/peku_unknown.bmp"
                },
                {
                  "id": 4,
                  "text": "dialog.bearingTypeListValues.radialRollerBearing",
                  "imageUrl": "https://mountingmanager-cae.schaeffler.com/api/Images/zyro_unknown.bmp"
                }
              ]
            }
          ]
        }
      ],
      "subPages": [
        {
          "id": "RSY_PAGE_BEARING_SERIES",
          "title": "RSY_BEARING_SERIES",
          "text": "dialog.bearing",
          "visible": {
            "operator": "NOT_EQUAL",
            "left": {
              "id": "RSY_BEARING_TYPE",
              "object": null,
              "specifications": [],
              "expressionType": "MEMBER"
            },
            "right": {
              "value": null,
              "dataType": "NONE",
              "expressionType": "CONSTANT"
            },
            "expressionType": "BINARY"
          },
          "groups": [
            {
              "members": [
                {
                  "id": "RSY_BEARING_SERIES",
                  "text": "dialog.bearingSeries",
                  "type": "lazy-list",
                  "optionsUrl": "OPTIONS_URL/bearing-types/<RSY_BEARING_TYPE>/series",
                  "sources": [
                    {
                      "id": "RSY_BEARING_TYPE",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    }
                  ],
                  "defaultValue": null,
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  }
                }
              ],
              "subPages": null
            }
          ]
        },
        {
          "id": "RSY_PAGE_BEARING",
          "title": "RSY_BEARING",
          "text": "dialog.bearing",
          "visible": {
            "operator": "AND",
            "left": {
              "operator": "NOT_EQUAL",
              "left": {
                "id": "RSY_BEARING_SERIES",
                "object": null,
                "specifications": [],
                "expressionType": "MEMBER"
              },
              "right": {
                "value": null,
                "dataType": "NONE",
                "expressionType": "CONSTANT"
              },
              "expressionType": "BINARY"
            },
            "right": {
              "operator": "NOT_EQUAL",
              "left": {
                "id": "RSY_BEARING_TYPE",
                "object": null,
                "specifications": [],
                "expressionType": "MEMBER"
              },
              "right": {
                "value": null,
                "dataType": "NONE",
                "expressionType": "CONSTANT"
              },
              "expressionType": "BINARY"
            },
            "expressionType": "BINARY"
          },
          "groups": [
            {
              "members": [
                {
                  "id": "RSY_BEARING",
                  "text": "dialog.bearing",
                  "type": "lazy-list",
                  "optionsUrl": "OPTIONS_URL/bearing-types/<RSY_BEARING_TYPE>/series/<RSY_BEARING_SERIES>/bearings",
                  "sources": [
                    {
                      "id": "RSY_BEARING_TYPE",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "RSY_BEARING_SERIES",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    }
                  ],
                  "defaultValue": null,
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  }
                }
              ],
              "subPages": null
            }
          ]
        }
      ]
    },
    {
      "id": "PAGE_MOUNTING_MANAGER_SEAT",
      "title": "IDMM_BEARING_SEAT",
      "text": "dialog.bearingSeat",
      "visible": {
        "value": true,
        "dataType": "boolean",
        "functionType": null,
        "expressionType": "CONSTANT",
        "ids": null,
        "id": null
      },
      "groups": [
        {
          "members": [
            {
              "id": "IDMM_BEARING_SEAT",
              "text": "dialog.bearingSeat",
              "type": "lazy-list",
              "defaultValue": null,
              "sources": [
                {
                  "id": "RSY_BEARING_TYPE",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                },
                {
                  "id": "RSY_BEARING_SERIES",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                },
                {
                  "id": "RSY_BEARING",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                }
              ],
              "optionsUrl": "OPTIONS_URL/bearing-types/<RSY_BEARING_TYPE>/series/<RSY_BEARING_SERIES>/bearings/<RSY_BEARING>",
              "abbreviation": "",
              "memberType": "Variable",
              "requiresUpdate": true,
              "requiresInitialization": false,
              "input": {
                "value": true,
                "dataType": "boolean",
                "functionType": null,
                "expressionType": "CONSTANT",
                "ids": null,
                "id": null
              }
            }
          ],
          "subPages": null
        }
      ]
    },
    {
      "id": "PAGE_MOUNTING_MANAGER_MEASURING_MOUTING_METHODS",
      "title": "IDMM_MEASSURING_METHOD",
      "text": "dialog.measuringAndMounting",
      "visible": {
        "value": true,
        "dataType": "boolean",
        "functionType": null,
        "expressionType": "CONSTANT",
        "ids": null,
        "id": null
      },
      "groups": [
        {
          "members": [
            {
              "id": "IDMM_MEASSURING_METHOD",
              "text": "dialog.measuringMethod",
              "type": "lazy-list",
              "defaultValue": null,
              "sources": [
                {
                  "id": "RSY_BEARING_TYPE",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                },
                {
                  "id": "RSY_BEARING_SERIES",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                },
                {
                  "id": "RSY_BEARING",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                },
                {
                  "id": "IDMM_BEARING_SEAT",
                  "object": null,
                  "specifications": [],
                  "expressionType": "MEMBER"
                }
              ],
              "optionsUrl": "OPTIONS_URL/bearing-types/<RSY_BEARING_TYPE>/series/<RSY_BEARING_SERIES>/bearings/<RSY_BEARING>/seats/<IDMM_BEARING_SEAT>/measuringMethods",
              "abbreviation": "",
              "memberType": "Variable",
              "requiresUpdate": true,
              "requiresInitialization": false,
              "input": {
                "value": true,
                "dataType": "boolean",
                "functionType": null,
                "expressionType": "CONSTANT",
                "ids": null,
                "id": null
              }
            }
          ]
        }
      ],
      "subPages": [
        {
          "id": "PAGE_MOUNTING_MANAGER_MOUTING_METHODS",
          "title": "IDMM_MOUNTING_METHOD",
          "text": "dialog.mounting",
          "visible": {
            "operator": "NOT_EQUAL",
            "left": {
              "id": "IDMM_MEASSURING_METHOD",
              "object": null,
              "specifications": [],
              "expressionType": "MEMBER"
            },
            "right": {
              "value": null,
              "dataType": "NONE",
              "expressionType": "CONSTANT"
            },
            "expressionType": "BINARY"
          },
          "groups": [
            {
              "members": [
                {
                  "id": "IDMM_MOUNTING_METHOD",
                  "text": "dialog.mountingMethod",
                  "type": "lazy-list",
                  "optionsUrl": "OPTIONS_URL/bearing-types/<RSY_BEARING_TYPE>/series/<RSY_BEARING_SERIES>/bearings/<RSY_BEARING>/seats/<IDMM_BEARING_SEAT>/measuringMethods/<IDMM_MEASSURING_METHOD>/mountingMethods",
                  "sources": [
                    {
                      "id": "RSY_BEARING_TYPE",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "RSY_BEARING_SERIES",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "RSY_BEARING",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "IDMM_BEARING_SEAT",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "IDMM_MEASSURING_METHOD",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    }
                  ],
                  "defaultValue": null,
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  },
                  "runtimeDescription": {
                    "assignMembers": [
                      {
                        "assignee": {
                          "id": "IDMM_HYDRAULIC_NUT_TYPE",
                          "object": null,
                          "specifications": [],
                          "expressionType": "MEMBER"
                        },
                        "initializer": {
                          "columnName": "IDMM_HYDRAULIC_NUT_TYPE",
                          "tableId": "TBL_BEARING_PREFLIGHT",
                          "defaultExpression": {
                            "id": "IDMM_HYDRAULIC_NUT_TYPE",
                            "object": null,
                            "specifications": [],
                            "expressionType": "MEMBER"
                          },
                          "whereConditions": [],
                          "functionType": "SELECT_STATEMENT",
                          "expressionType": "FUNCTION"
                        }
                      },
                      {
                        "assignee": {
                          "id": "IDMM_INNER_RING_EXPANSION",
                          "object": null,
                          "specifications": [],
                          "expressionType": "MEMBER"
                        },
                        "initializer": {
                          "columnName": "IDMM_INNER_RING_EXPANSION",
                          "tableId": "TBL_BEARING_PREFLIGHT",
                          "defaultExpression": {
                            "id": "IDMM_INNER_RING_EXPANSION",
                            "object": null,
                            "specifications": [],
                            "expressionType": "MEMBER"
                          },
                          "whereConditions": [],
                          "functionType": "SELECT_STATEMENT",
                          "expressionType": "FUNCTION"
                        }
                      },
                      {
                        "assignee": {
                          "id": "IDMM_RADIAL_CLEARANCE_REDUCTION",
                          "object": null,
                          "specifications": [],
                          "expressionType": "MEMBER"
                        },
                        "initializer": {
                          "columnName": "IDMM_RADIAL_CLEARANCE_REDUCTION",
                          "tableId": "TBL_BEARING_PREFLIGHT",
                          "defaultExpression": {
                            "id": "IDMM_RADIAL_CLEARANCE_REDUCTION",
                            "object": null,
                            "specifications": [],
                            "expressionType": "MEMBER"
                          },
                          "whereConditions": [],
                          "functionType": "SELECT_STATEMENT",
                          "expressionType": "FUNCTION"
                        }
                      }
                    ]
                  }
                }
              ]
            }
          ]
        }
      ]
    },
    {
      "id": "PROPERTY_PAGE_MOUNTING_SITUATION",
      "title": "PROPERTY_PAGE_TITLE_MOUNTING_SITUATION",
      "text": "dialog.calculationOptions",
      "visible": {
        "value": false,
        "dataType": null,
        "functionType": "SET_ID_CHECK",
        "expressionType": "FUNCTION",
        "ids": ["LB_AXIAL_DISPLACEMENT"],
        "id": {
          "id": "IDMM_MEASSURING_METHOD",
          "object": null,
          "specifications": [],
          "expressionType": "MEMBER"
        }
      },
      "groups": [
        {
          "members": []
        }
      ],
      "subPages": [
        {
          "id": "PROPERTY_PAGE_MOUNTING",
          "title": "STRING_OUTP_MOUNTING",
          "text": "dialog.mounting",
          "visible": {
            "value": true,
            "dataType": "boolean",
            "functionType": null,
            "expressionType": "CONSTANT",
            "ids": null,
            "id": null
          },
          "groups": [
            {
              "members": [
                {
                  "id": "IDMM_HYDRAULIC_NUT_TYPE",
                  "text": "dialog.hydraulicNutType",
                  "type": "lazy-list",
                  "unit": "",
                  "precision": null,
                  "defaultValue": "LB_NOT_AVAILABLE",
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  },
                  "sources": [
                    {
                      "id": "RSY_BEARING_TYPE",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "RSY_BEARING_SERIES",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "RSY_BEARING",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    {
                      "id": "IDMM_BEARING_SEAT",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    }
                  ],
                  "optionsUrl": "OPTIONS_URL/bearing-preflight"
                },
                {
                  "id": "IDMM_NUMBER_OF_PREVIOUS_MOUNTINGS",
                  "text": "dialog.previousMountings",
                  "type": "list",
                  "unit": "",
                  "precision": null,
                  "defaultValue": "LB_ZERO_TO_ONE",
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": null,
                  "minEqual": null,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  },
                  "listValues": [
                    {
                      "id": "LB_ZERO_TO_ONE",
                      "text": "dialog.previousMountingsListValues.zeroToOne"
                    },
                    {
                      "id": "LB_TWO",
                      "text": "dialog.previousMountingsListValues.two"
                    },
                    {
                      "id": "LB_THREE",
                      "text": "dialog.previousMountingsListValues.three"
                    },
                    {
                      "id": "LB_FOUR",
                      "text": "dialog.previousMountingsListValues.four"
                    },
                    {
                      "id": "LB_FIVE",
                      "text": "dialog.previousMountingsListValues.five"
                    },
                    {
                      "id": "LB_SIX_PLUS",
                      "text": "dialog.previousMountingsListValues.sixPlus"
                    }
                  ]
                },
                {
                  "id": "IDMM_CLEARANCE_REDUCTION_INPUT",
                  "text": "dialog.mountingOptions",
                  "type": "list",
                  "unit": "",
                  "precision": null,
                  "defaultValue": "LB_RADIAL_CLEARANCE_REDUCTION",
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": null,
                  "minEqual": null,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  },
                  "listValues": [
                    {
                      "id": "LB_RADIAL_CLEARANCE_REDUCTION",
                      "text": "dialog.mountingOptionsListValues.radialClearanceReduction"
                    },
                    {
                      "id": "LB_INNER_RING_EXPANSION",
                      "text": "dialog.mountingOptionsListValues.innerRingExpansion"
                    }
                  ]
                },
                {
                  "id": "IDMM_INNER_RING_EXPANSION",
                  "text": "dialog.innerRingExpansion",
                  "type": "number",
                  "unit": "‰",
                  "precision": 3,
                  "defaultValue": 0.0,
                  "abbreviation": "eps_d_ir",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": null,
                  "minEqual": null,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "operator": "EQUAL",
                    "left": {
                      "id": "IDMM_CLEARANCE_REDUCTION_INPUT",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    "right": {
                      "value": "LB_INNER_RING_EXPANSION",
                      "dataType": "NONE",
                      "expressionType": "CONSTANT"
                    },
                    "expressionType": "BINARY"
                  }
                },
                {
                  "id": "IDMM_RADIAL_CLEARANCE_REDUCTION",
                  "text": "dialog.radialClearanceReduction",
                  "type": "number",
                  "unit": "µm",
                  "precision": 1,
                  "defaultValue": 0.0,
                  "abbreviation": "delta_sr",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": null,
                  "minEqual": null,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "operator": "EQUAL",
                    "left": {
                      "id": "IDMM_CLEARANCE_REDUCTION_INPUT",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    "right": {
                      "value": "LB_RADIAL_CLEARANCE_REDUCTION",
                      "dataType": "NONE",
                      "expressionType": "CONSTANT"
                    },
                    "expressionType": "BINARY"
                  }
                }
              ],
              "subPages": []
            }
          ],
          "helpId": null
        },
        {
          "id": "PROPERTY_PAGE_MOUNTING_SITUATION_SUB",
          "title": "STRING_OUTP_SHAFT_DATA",
          "text": "dialog.shaftData",
          "visible": {
            "value": true,
            "dataType": "boolean",
            "functionType": null,
            "expressionType": "CONSTANT",
            "ids": null,
            "id": null
          },
          "groups": [
            {
              "members": [
                {
                  "id": "IDMM_INNER_SHAFT_DIAMETER",
                  "text": "dialog.innerShaftDiameter",
                  "type": "number",
                  "unit": "mm",
                  "precision": 3,
                  "defaultValue": 0.0,
                  "abbreviation": "d_sft",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": null,
                  "minEqual": null,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  },
                  "listValues": []
                },
                {
                  "id": "IDMM_SHAFT_MATERIAL",
                  "text": "dialog.shaftMaterial",
                  "type": "list",
                  "unit": "",
                  "precision": null,
                  "defaultValue": "STEEL_20_DEGREE",
                  "abbreviation": "",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": null,
                  "minEqual": null,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "value": true,
                    "dataType": "boolean",
                    "functionType": null,
                    "expressionType": "CONSTANT",
                    "ids": null,
                    "id": null
                  },
                  "listValues": [
                    {
                      "id": "STEEL_20_DEGREE",
                      "text": "dialog.shaftMaterialListValues.steel"
                    },
                    {
                      "id": "STEEL_100_CR_6",
                      "text": "dialog.shaftMaterialListValues.100Cr6"
                    },
                    {
                      "id": "STEEL_C_45",
                      "text": "dialog.shaftMaterialListValues.C45"
                    },
                    {
                      "id": "STEEL_16_MN_CR_5",
                      "text": "dialog.shaftMaterialListValues.16MnCr5"
                    },
                    {
                      "id": "CAST_IRON_GG14",
                      "text": "dialog.shaftMaterialListValues.castIronGG14"
                    },
                    {
                      "id": "CAST_IRON_GG26",
                      "text": "dialog.shaftMaterialListValues.castIronGG26"
                    },
                    {
                      "id": "CAST_IRON_GGG",
                      "text": "dialog.shaftMaterialListValues.castIronGGG"
                    },
                    {
                      "id": "BRASS",
                      "text": "dialog.shaftMaterialListValues.brass"
                    },
                    {
                      "id": "BRONZE",
                      "text": "dialog.shaftMaterialListValues.bronze"
                    },
                    {
                      "id": "ALLOY_ALUMINIUM",
                      "text": "dialog.shaftMaterialListValues.alloyAluminium"
                    },
                    {
                      "id": "USER_DEFINED_MATERIAL",
                      "text": "dialog.shaftMaterialListValues.userDefinedMaterial"
                    }
                  ],
                  "runtimeDescription": {
                    "assignMembers": [
                      {
                        "assignee": {
                          "id": "IDMM_MODULUS_OF_ELASTICITY",
                          "object": null,
                          "specifications": [],
                          "expressionType": "MEMBER"
                        },
                        "initializer": {
                          "columnName": "IDMM_MODULUS_OF_ELASTICITY",
                          "tableId": "TBL_SHAFT_MATERIAL",
                          "defaultExpression": {
                            "id": "IDMM_MODULUS_OF_ELASTICITY",
                            "object": null,
                            "specifications": [],
                            "expressionType": "MEMBER"
                          },
                          "whereConditions": [],
                          "functionType": "SELECT_STATEMENT",
                          "expressionType": "FUNCTION"
                        }
                      },
                      {
                        "assignee": {
                          "id": "IDMM_POISSON_RATIO",
                          "object": null,
                          "specifications": [],
                          "expressionType": "MEMBER"
                        },
                        "initializer": {
                          "columnName": "IDMM_POISSON_RATIO",
                          "tableId": "TBL_SHAFT_MATERIAL",
                          "defaultExpression": {
                            "id": "IDMM_POISSON_RATIO",
                            "object": null,
                            "specifications": [],
                            "expressionType": "MEMBER"
                          },
                          "whereConditions": [],
                          "functionType": "SELECT_STATEMENT",
                          "expressionType": "FUNCTION"
                        }
                      }
                    ]
                  }
                },
                {
                  "id": "IDMM_MODULUS_OF_ELASTICITY",
                  "text": "dialog.modulusOfElasticity",
                  "type": "number",
                  "unit": "N/mm²",
                  "precision": 0,
                  "defaultValue": 210000.0,
                  "abbreviation": "E",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": 0.0,
                  "minEqual": false,
                  "maxValue": null,
                  "maxEqual": null,
                  "input": {
                    "operator": "EQUAL",
                    "left": {
                      "id": "IDMM_SHAFT_MATERIAL",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    "right": {
                      "value": "USER_DEFINED_MATERIAL",
                      "dataType": "NONE",
                      "expressionType": "CONSTANT"
                    },
                    "expressionType": "BINARY"
                  },
                  "listValues": []
                },
                {
                  "id": "IDMM_POISSON_RATIO",
                  "text": "dialog.poissonRatio",
                  "type": "number",
                  "unit": "",
                  "precision": 2,
                  "defaultValue": 0.3,
                  "abbreviation": "Nue",
                  "memberType": "Variable",
                  "requiresUpdate": true,
                  "requiresInitialization": false,
                  "minValue": 0.0,
                  "minEqual": false,
                  "maxValue": 0.51,
                  "maxEqual": true,
                  "input": {
                    "operator": "EQUAL",
                    "left": {
                      "id": "IDMM_SHAFT_MATERIAL",
                      "object": null,
                      "specifications": [],
                      "expressionType": "MEMBER"
                    },
                    "right": {
                      "value": "USER_DEFINED_MATERIAL",
                      "dataType": "NONE",
                      "expressionType": "CONSTANT"
                    },
                    "expressionType": "BINARY"
                  },
                  "listValues": []
                }
              ],
              "subPages": []
            }
          ]
        }
      ]
    },
    {
      "id": "PAGE_RESULT",
      "title": "IDMM_RESULT",
      "text": "dialog.report",
      "visible": {
        "value": false
      },
      "groups": []
    }
  ]
}
