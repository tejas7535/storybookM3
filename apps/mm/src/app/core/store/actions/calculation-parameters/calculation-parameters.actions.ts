import { createAction, props } from '@ngrx/store';

import { CalculationParameters } from '../../models/calculation-parameters-state.model';

export const setCalculationParameters = createAction(
  '[Calculation Parameters] Set Calculation Parameters',
  props<{ parameters: CalculationParameters }>()
);
