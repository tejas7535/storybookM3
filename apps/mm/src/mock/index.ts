import { BearinxDialogDescription } from '@caeonline/dynamic-forms';

import { environment } from '../environments/environment';
import * as dialog from './dialog.json';

const envDialogString = JSON.stringify(dialog).replace(
  /OPTIONS_URL/g,
  environment.baseUrl
);

const envDialog = JSON.parse(envDialogString);

export const DIALOG = (envDialog as any).default as BearinxDialogDescription;
