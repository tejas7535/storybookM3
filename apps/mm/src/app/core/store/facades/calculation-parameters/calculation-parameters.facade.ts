import { Injectable } from '@angular/core';

import { Store } from '@ngrx/store';

import { CalculationParametersActions } from '../../actions/calculation-parameters';
import { CalculationParameters } from '../../models/calculation-parameters-state.model';
import { CalculationParametersSelector } from '../../selectors';

@Injectable({
  providedIn: 'root',
})
export class CalculationParametersFacade {
  public readonly getCalculationParameters$ = this.store.select(
    CalculationParametersSelector.getCalculationParameters
  );

  constructor(private readonly store: Store) {}

  setCalculationParameters(formProperties: CalculationParameters): void {
    this.store.dispatch(
      CalculationParametersActions.setCalculationParameters({
        parameters: formProperties,
      })
    );
  }
}
