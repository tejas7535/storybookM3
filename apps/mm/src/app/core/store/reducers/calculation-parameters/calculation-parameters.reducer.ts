import { createReducer, on } from '@ngrx/store';

import { CalculationParametersActions } from '../../actions/calculation-parameters';
import { CalculationParametersState } from '../../models/calculation-parameters-state.model';

export const initialState: CalculationParametersState = {
  parameters: undefined,
};

export const calculationParametersReducer = createReducer(
  initialState,
  on(
    CalculationParametersActions.setCalculationParameters,
    (state, { parameters }): CalculationParametersState => ({
      ...state,
      parameters,
    })
  )
);
