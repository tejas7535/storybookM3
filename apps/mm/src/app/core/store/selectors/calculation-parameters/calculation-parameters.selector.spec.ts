import { APP_STATE_MOCK } from '../../../../../testing/mocks/store/app-state.mock';
import { CalculationParametersState } from '../../models/calculation-parameters-state.model';
import { getCalculationParameters } from './calculation-parameters.selector';

describe('CalculationParametersSelector', () => {
  const state: CalculationParametersState = {
    ...APP_STATE_MOCK.calculationParameters,
  };

  describe('getCalculationParameters', () => {
    describe('when parameters are not defined', () => {
      it('should return undefined', () => {
        expect(getCalculationParameters(state)).toBeUndefined();
      });
    });

    describe('when parameters are defined', () => {
      it('should return inputs', () => {
        expect(
          getCalculationParameters({
            calculationParameters: {
              parameters: {
                RSY_BEARING_SERIES: 'some series',
              },
            },
          })
        ).toMatchSnapshot();
      });
    });
  });
});
