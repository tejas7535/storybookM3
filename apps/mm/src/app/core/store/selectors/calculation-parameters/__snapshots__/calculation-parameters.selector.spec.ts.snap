// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationParametersSelector getCalculationParameters when parameters are defined should return inputs 1`] = `
Object {
  "RSY_BEARING_SERIES": "some series",
}
`;
